lab21/exercise_3.py

```diff
@@ -1,19 +1,29 @@
+from __future__ import annotations
 class Widget:
-
     def __init__(self, name):
         self.name = name
         self.built = False
         self.dependencies = []
 
-    def add_dependency(self, *arg):
-        for d in arg:
-            self.dependencies.append(d)
+    # Using type hints and += we can simplify this method
+    # Please watch the type hints lesson here for more info:
+    # https://www.tocode.co.il/bundles/advanced-python3/lessons/type-hints
+    # Note that because I use Widget as a type hint, and it hasn't been fully defined yet
+    # (we're in Widget class definitions now)
+    # I had to add the "from __future__ import annotations" statement to the
+    # top of the file
+    def add_dependency(self, *widgets: Widget):
+        self.dependencies += widgets
 
     def build(self):
         self.built = True
         for dependency in self.dependencies:
-            if not dependency.built:
-                dependency.build()
+            # I prefer to have the "important" part aligned with the loop,
+            # so I would write:
+            if dependency.built: continue
```

Comment:
Personally I don't like to use `break` and `continue`. I try to avoid them. I also usually don't use `return` inside a for loop.